autotest/ogr/ogr_csv.py

```diff
@@ -3016,6 +3016,74 @@ def test_ogr_csv_read_header_with_line_break():
     ]
 
 
+###############################################################################
+# Test geometry coordinate precision support
+
+
+@pytest.mark.parametrize("geometry_format", ["AS_WKT", "AS_XYZ"])
+def test_ogr_csv_geom_coord_precision(tmp_vsimem, geometry_format):
+
+    filename = str(tmp_vsimem / "test.csv")
+    ds = gdal.GetDriverByName("CSV").Create(filename, 0, 0, 0, gdal.GDT_Unknown)
+    geom_fld = ogr.GeomFieldDefn("geometry", ogr.wkbUnknown)
+    prec = ogr.CreateGeomCoordinatePrecision()
+    prec.Set(1e-5, 1e-3, 1e-2)
+    geom_fld.SetCoordinatePrecision(prec)
+    lyr = ds.CreateLayerFromGeomFieldDefn(
+        "test", geom_fld, ["GEOMETRY=" + geometry_format]
+    )
+    geom_fld = lyr.GetLayerDefn().GetGeomFieldDefn(0)
+    prec = geom_fld.GetCoordinatePrecision()
+    assert prec.GetXYResolution() == 1e-5
+    assert prec.GetZResolution() == 1e-3
+    assert prec.GetMResolution() == 1e-2
+    f = ogr.Feature(lyr.GetLayerDefn())
+    f.SetGeometry(
+        ogr.CreateGeometryFromWkt("POINT(1.23456789 2.34567891 9.87654321 -1.23456789)")
+    )
+    lyr.CreateFeature(f)
+    ds.Close()
+
+    f = gdal.VSIFOpenL(filename, "rb")
+    assert f
+    data = gdal.VSIFReadL(1, 10000, f)
+    gdal.VSIFCloseL(f)
+
+    if geometry_format == "AS_WKT":
+        assert b"POINT ZM (1.23457 2.34568 9.877 -1.23)" in data
+    else:
+        assert b"1.23457,2.34568,9.877" in data
+
+
+###############################################################################
+# Test geometry coordinate precision support
+
+
+@pytest.mark.require_geos
+def test_ogr_csv_geom_coord_precision_OGR_APPLY_GEOM_SET_PRECISION(tmp_vsimem):
+
+    filename = str(tmp_vsimem / "test.csv")
+    ds = gdal.GetDriverByName("CSV").Create(filename, 0, 0, 0, gdal.GDT_Unknown)
+    geom_fld = ogr.GeomFieldDefn("geometry", ogr.wkbUnknown)
+    prec = ogr.CreateGeomCoordinatePrecision()
+    prec.Set(0.5, 0, 0)
+    geom_fld.SetCoordinatePrecision(prec)
+    lyr = ds.CreateLayerFromGeomFieldDefn("test", geom_fld, ["GEOMETRY=AS_WKT"])
+    f = ogr.Feature(lyr.GetLayerDefn())
+    f.SetGeometry(
+        ogr.CreateGeometryFromWkt("POLYGON((0 0,0.5 0.4,1 0,1 1,0.5 0.6,0 1,0 0))")
+    )
+    with gdaltest.config_option("OGR_APPLY_GEOM_SET_PRECISION", "YES"):
+        lyr.CreateFeature(f)
+    ds.Close()
+
+    f = gdal.VSIFOpenL(filename, "rb")
+    assert f
+    data = gdal.VSIFReadL(1, 10000, f)
+    gdal.VSIFCloseL(f)
+    assert b"MULTIPOLYGON" in data
```

Comment:
I suspect some irony in your comment ;-) Well, the test is actually smart, but without comments, it is admitedly hard to realize.
I've added the following ones:
```
     geom_fld.SetCoordinatePrecision(prec)
     lyr = ds.CreateLayerFromGeomFieldDefn("test", geom_fld, ["GEOMETRY=AS_WKT"])
     f = ogr.Feature(lyr.GetLayerDefn())
+    # We create an initial polygon, which is valid if the precision is infinite,
+    # but when rounding coordinates to 0.5 resolution, it would become invalid.
     f.SetGeometry(
         ogr.CreateGeometryFromWkt("POLYGON((0 0,0.5 0.4,1 0,1 1,0.5 0.6,0 1,0 0))")
     )
@@ -3081,6 +3083,9 @@ def test_ogr_csv_geom_coord_precision_OGR_APPLY_GEOM_SET_PRECISION(tmp_vsimem):
     assert f
     data = gdal.VSIFReadL(1, 10000, f)
     gdal.VSIFCloseL(f)
+    # We just check that GEOS did its job by turning the polygon into a
+    # multipolygon made of 2 parts. To avoid being dependent on GEOS version,
+    # we just check for the MULTIPOLYGON keyword.
     assert b"MULTIPOLYGON" in data
 ```

Reply:
No, I wasn't ironic, it took me a while to visualize the degeneration of the single polygon into two due to the rounding and I thought it was smart. But yes, a comment can be helpful.